### DISCUSSION TITLE
ANGULAR __name is not defined

### DISCUSSION THREAD

**Discussion by amoscatelli** [google/model-viewer#3190] [Q&A] [answered]
Hi there !
I integrated model-viewer with an angular application.
The integration brooke without any logical reason in my test environment :

https://showcase.test.divelitalia.it/it/products/sole/fashion/glamourmask/viewer

Uncaught ReferenceError: __name is not defined
    at 535b2e5b-a889-4af1-8ba5-6b771dba1837:137:581
    
The error occurs inside the /* worker */ javascript.

The most suspicious thing is that locally I can successfully compile the project with test environment and serving generated files works without any problem.

The only relevat change I made recently is that I migrated my Jenkins to arm64 architecture.
Is there any known problem related to model-viewer or ThreeJS or DRACOLoader and arm64 that could explain such a problem ?

The version I am using is this
"@google/model-viewer": "1.10.1"
But I think this is not related, sinced it used to work with the same and previous versions.

Please help, I really dunno what is happening here.
Thank you in advance.

**Comment by amoscatelli** [comment]
I confirm it's not related to arm64 architecture compiling the project.
Still investigating.

**Reply by elalish** [reply]
These kinds of errors are generally not in MV's code. However, this is a good place to find people who have worked through similar issues with frameworks like Angular.

**Reply by axwellllll** [reply]
do you have example of the code pls?

**Comment by amoscatelli** [comment, marked as answer]
I found out what caused the issue.
The issue is related to a specific gltf requiring a draco functionality.
This triggers the library to use DRACOLoader.js from three-js.
In this scenario, the ONLY WAY I found to use @google/model-viewer in an angular project is to copy model-viewer.min.js into assets folder inside angular.json, and import it from assets folder in index.html (or you can use the remote import directly in index.html).
In any other way the DRACOWorker declaration is changed by angular compiler at compilation time and __name error is triggered.

The breaking change, maybe, was the migration to angular 13 we also did a month ago, and enabling ivy compiler.
I am not sure why it still works compiling on my machine while directly importing the library in typescript, but my test environment is ok now.

So here is the summary 

NOT WORKING:
Importing @google/model-viewer inside polyfills.ts
Importing @google/model-viewer inside a component 
Declaring model-viewer.min.js (umd version too) among scripts in angular.json

WORKING
Importing remote model-viewer.min.js inside index.html
Copying model-viewer.min.js in assets inside angular.json and importing model-viewer.min.js from assets in index.html
